debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/WAL2JSONColumn.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.postgresql.connection.wal2json;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.postgresql.core.Oid;
+import org.postgresql.geometric.PGpoint;
+import org.postgresql.jdbc.PgArray;
+import org.postgresql.util.PGInterval;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.postgresql.PostgresValueConverter;
+import io.debezium.connector.postgresql.RecordsStreamProducer.PgConnectionSupplier;
+import io.debezium.connector.postgresql.connection.ReplicationMessage;
+import io.debezium.document.Value;
+import io.debezium.relational.Column;
+
+/**
+ * Logical encapsulation of column changes sent by <a href="https://github.com/debezium/postgres-decoderbufs">Postgres Decoderbufs</>
+ * 
+ * @author Jiri Pechanec
+ *
+ */
+class WAL2JSONColumn extends ReplicationMessage.Column {
+    private static final Logger logger = LoggerFactory.getLogger(WAL2JSONColumn.class);
+
+    private final Value rawValue;
+
+    public WAL2JSONColumn(final String name, final String type, final Value value) {
+        super(name, type);
+        this.rawValue = value;
+    }
+
+    /**
+     * Converts the value (string representation) coming from wal2json plugin to
+     * a Java value based on the type of the column from the message. This value will be converted later on if necessary by the
+     * {@link PostgresValueConverter#converter(Column, Field)} instance to match whatever the Connect schema type expects.
+     *
+     * Note that the logic here is tightly coupled (i.e. dependent) on the wal2json plugin logic which writes the actual
+     * Protobuf messages.
+     *
+     * @param a supplier to get a connection to Postgres instance for array handling
+     * @return the value; may be null
+     */
+    @Override
+    public Object getValue(final PgConnectionSupplier connection) {
+        String columnType = (String)getType();
+        switch (columnType) {
+            case "bool":
+                return rawValue.isNotNull() ? rawValue.asBoolean() : null;
+            case "int2":
+            case "int4":
+                return rawValue.isNotNull() ? rawValue.asInteger() : null;
+            case "int8":
+            case "oid":
+                return rawValue.isNotNull() ? rawValue.asLong() : null;
+            case "float4":
+                return rawValue.isNotNull() ? rawValue.asFloat() : null;
+            case "float8":
+                return rawValue.isNotNull() ? rawValue.asDouble() : null;
+            case "numeric":
```

Comment:
In case of `float8` it is an equivalent of `double`. In case of `numeric` we are losing precision but we are consitent for both decoders. I prefer to stay consistent and fix precision problem DBZ-351 for both connectors at the same time.

Reply:
Why do it first wrong though if we can do it correctly for this plug-in from the get-go? I'd buy the consistency argument if there was some user-facing option or so involved, but it isn't. So users won't benefit from doing it consistently wrong IMO.